16_interpretable-machine-learning.Rmd

```diff
@@ -1,13 +1,464 @@
 # Interpretable Machine Learning
 
+```{r, setup, include=FALSE}
+knitr::opts_chunk$set(
+  eval = FALSE
+)
+```
+
 **Learning objectives:**
 
-- THESE ARE NICE TO HAVE BUT NOT ABSOLUTELY NECESSARY
+1. Understand the importance of interpretability in machine learning models.
+2. Learn about global and local interpretation.
+3. Understand the trade-off in interpretation.
+4. Learn about model-specific implementation.
+5. Understand the concept of permutation-based feature importance.
+6. Understand the concept of partial dependence.
+7. Learn about Individual Conditional Expectation (ICE).
+
+## Introduction
+
+Model's **interpretability** is crucial for:
+
+- Business adoption
+- Model documentation
+- Regulatory oversight
+- Human acceptance and trust
+
+**Interpretable Machine Learning** provides solutions to explain at different levels.
+
+- **Global Interpretation** explain  how the model makes predictions, based on a holistic view of its **features**. They can explain:
+  - which features are the most influential (via **feature importance**)
+  - How the most influential variables drive the model output (via **feature effects**)
+
+- **Local Interpretation** becomes very important when the effect of most influential model's features doesn't the biggest influence for a particular observation as it helps us to understand what **features are influencing the predicted response** for a given observation or an small group of observations.
+  - Local interpretable model-agnostic explanations (LIME)
+  - Shapley values
+  - Localized step-wise procedures
+
+## Setting enviroment
+
+### Loading libraries
+
+```{r libraries, message=FALSE, echo=TRUE, results='hide'}
```

Comment:
```suggestion
```{r 16-libraries, message=FALSE, echo=TRUE, results='hide'}
```

Reply:
Thanks, I didn't know that